app/gather/api/views.py

```diff
@@ -44,3 +60,37 @@ class MaskViewSet(MtViewSetMixin, ModelViewSet):
     search_fields = ('survey__name', 'name', 'columns',)
     ordering = ('survey', 'name',)
     mt_field = 'survey'
+
+
+@api_view(['POST', 'DELETE'])
+@renderer_classes([JSONRenderer])
+@permission_classes([permissions.IsAuthenticated])
+def consumer_config(request, *args, **kwargs):
+    _survey_name = request.data.get('name')
+    _headers = {}
+    if not _survey_name:
+        return Response('Missing survey name', status=status.HTTP_400_BAD_REQUEST)
+
+    _survey_name_clean = __clean_name(_survey_name)
+
+    _realm = get_current_realm(request)
+    _headers[settings.TENANCY_HEADER] = _realm
+
+    # Configure Consumers
+    if settings.AUTO_CONFIG_CONSUMERS:
+        consumer_settings = copy.deepcopy(settings.CONSUMER_SETTINGS)
+        if request.method == 'POST':
+            consumers, errors = configure_consumers(consumer_settings, _survey_name_clean, _headers)
+            if errors:
+                return Response(errors, status=status.HTTP_400_BAD_REQUEST)
+            return Response(f'Configured {consumers} consumers successfully', status=status.HTTP_200_OK)
+        elif request.method == 'DELETE':
+            consumer_settings = copy.deepcopy(settings.CONSUMER_SETTINGS)
+            errors = delete_survey_subscription(consumer_settings, _survey_name_clean, _headers)
+            if errors:
+                return Response(errors, status=status.HTTP_400_BAD_REQUEST)
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        else:
+            pass    # pragma: no cover
```

Comment:
:thinking: 

Do you need it?

Reply:
flake8 100% coverage

Reply:
if you add `pragma: no cover` obviously :stuck_out_tongue: 

If you have `@api_view(['POST', 'DELETE'])` there is no else after `if request.method == 'POST': ... elif request.method == 'DELETE': ...`

Reply:
What I mean...

```python
@api_view(['POST', 'DELETE'])
@renderer_classes([JSONRenderer])
@permission_classes([permissions.IsAuthenticated])
def consumer_config(request, *args, **kwargs):
    # ...

    # Configure Consumers
    if not settings.AUTO_CONFIG_CONSUMERS:
        return Response('Consumer auto configuration is turned off', status=status.HTTP_200_OK)

    consumer_settings = copy.deepcopy(settings.CONSUMER_SETTINGS)
    if request.method == 'POST':
        # ...

    elif request.method == 'DELETE':
        # ...
```